conda_smithy/schema.py

```diff
@@ -0,0 +1,1328 @@
+
+
+class BotConfigSkipRenderChoices(str, Enum):
+    GITIGNORE = ".gitignore"
+    GITATTRIBUTES = ".gitattributes"
+    README = "README.md"
+    LICENSE = "LICENSE.txt"
+    GITHUB_WORKFLOWS = ".github/workflows"
+
+
+class BotConfigInspectionChoice(str, Enum):
+    HINT = "hint"
+    HINT_ALL = "hint-all"
+    HINT_SOURCE = "hint-source"
+    HINT_GRAYSKULL = "hint-grayskull"
+    UPDATE_ALL = "update-all"
+    UPDATE_SOURCE = "update-source"
+    UPDATE_GRAYSKULL = "update-grayskull"
+
+
+##############################################
+########## Model definitions #################
+##############################################
+
+
+class AzureSelfHostedRunnerSettings(BaseModel):
+    """This is the settings for self-hosted runners."""
+
+    pool: Optional[Dict[str, str]] = Field(
+        default_factory=lambda: {"vmImage": "ubuntu-latest"},
+        description="The pool of self-hosted runners, e.g. 'vmImage': 'ubuntu-latest'",
+    )
+
+    swapfile_size: Optional[Union[str, Nullable]] = Field(
+        default=None, description="Swapfile size in GiB"
+    )
+
+    timeoutInMinutes: Optional[int] = Field(
+        default=360, description="Timeout in minutes for the job"
+    )
+
+    variables: Optional[Dict[str, str]] = Field(
+        default_factory=dict, description="Variables"
+    )
+
+
+class AzureConfig(BaseModel):
+    """
+    This dictates the behavior of the Azure Pipelines CI service. It is a sub-mapping for Azure-specific configuration options. For more information and some variables specifications, see the [Azure Pipelines schema reference documentation](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/?view=azure-pipelines).
+    """
+
+    force: Optional[bool] = Field(
+        default=False,
+        description="Force building all supported providers",
+    )
+
+    free_disk_space: Optional[Union[bool, Nullable]] = Field(
+        default=None,
+        description="Free up disk space before running the Docker container for building on Linux",
+        exclude=True,  # Will not be rendered in the model dump
+    )
+
+    max_parallel: Optional[int] = Field(
+        default=50,
+        description="Limit the amount of CI jobs running concurrently at a given time",
+    )
+
+    project_id: Optional[str] = Field(
+        default="84710dde-1620-425b-80d0-4cf5baca359d",
+        description="The ID of the Azure Pipelines project",
+    )
+
+    project_name: Optional[str] = Field(
+        default="feedstock-builds",
+        description="The name of the Azure Pipelines project",
+    )
+
+    #########################################
+    ##### Self-hosted runners settings ######
+    #########################################
+    settings_linux: AzureSelfHostedRunnerSettings = Field(
+        default_factory=lambda: AzureSelfHostedRunnerSettings(
+            swapfile_size="0GiB"
+        ),
+        description="Linux-specific settings for self-hosted runners",
+    )
```

Comment:
This is not specific to self-hosted runners. It supports all options in https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/jobs-job?view=azure-pipelines which includes self-hosted runner implementation details.